src/main.c

```diff
@@ -386,12 +387,31 @@ static void listener_setup_ssl_add_host(struct listener_ssl_config_t *ssl_config
     ssl_config->hostnames.entries[ssl_config->hostnames.size++] = h2o_iovec_init(host.base, host_end - host.base);
 }
 
+static h2o_iovec_t *build_http2_origin_frame(const h2o_iovec_t *origins, size_t nr_origins)
```

Comment:
Would it make sense to change the signature of this function to: `(h2o_configurator_command_t *cmd, yoml_t *origins, size_t nr_origins)`?

If the argument to the directive was a scalar, the caller can just call `build_http2_origin_frame(cmd, value, 1)`. If the argument to the directive was a sequence, the caller should call `build_http2_origin_frame(cmd, value.data.sequence.elements, value.data.sequence.size)`.